src/ipc/ipc-common.c

```diff
@@ -117,22 +116,46 @@ struct ipc_comp_dev *ipc_get_ppl_comp(struct ipc *ipc, uint32_t pipeline_id, int
 		/* first try to find the module in the pipeline */
 		if (dev_comp_pipe_id(icd->cd) == pipeline_id) {
 			struct list_item *buffer_list = comp_buffer_list(icd->cd, dir);
+			bool b_endpoint = true;
```

Comment:
I'm not sure I follow this correctly. We use "endpoint" term in copier.c but you don't refer to that here? So this is just way to refer to the "edge" node or "terminal" of the pipe, the first/last component of a pipeline. Given the complexity of the code, would be good to not overload terms.

Reply:
@kv2019i , this function is comments as
"Walks through the list of components looking for a sink/source endpoint component of the given pipeline", so I use endpoint.

And @btian1  is working on copier and the endpoint will be remove soon.